From 66cdb9d9e167e1adc8eb2fc6e9485fcf7a0fcaef Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Wed, 24 Oct 2018 11:12:19 +0530
Subject: [PATCH 0480/1851] drm: xlnx: sdi: Disable bridge instance

commit 0ccf0d5f6f4b9c0d9156bffbb037304692f18518 from
https://github.com/Xilinx/linux-xlnx.git

This patch disables the bridge instance when sdi controller is disabled.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index 0e1dd6505ab2..25412dda5293 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -959,6 +959,9 @@ static void xlnx_sdi_disable(struct drm_encoder *encoder)
 {
 	struct xlnx_sdi *sdi = encoder_to_sdi(encoder);
 
+	if (sdi->bridge)
+		xlnx_bridge_disable(sdi->bridge);
+
 	xlnx_sdi_set_display_disable(sdi);
 	xlnx_stc_disable(sdi->base);
 }
-- 
2.31.1

